java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java

```diff
@@ -143,55 +143,69 @@ public Class<?> getClass1() {
     }
     
     private void buildType(){
-            TypeMirror tm = null;
-            VariableElement var = Utils.getField(parent.getTypeElement(), name);
-            if(var == null){
-                ExecutableElement acc = Utils.getAccesor(parent.getTypeElement(), name);
-                if(acc != null){
-                    tm = acc.getReturnType();
-                }
-            } else {
-                tm = var.asType();
+        TypeMirror tm = null;
+        VariableElement var = Utils.getField(parent.getTypeElement(), name);
+        if(var == null){
+            ExecutableElement acc = Utils.getAccesor(parent.getTypeElement(), name);
+            if(acc != null){
+                tm = acc.getReturnType();
             }
-            if( tm!=null ){
-                if(tm.getKind() == TypeKind.DECLARED){
-                    DeclaredType declaredType = (DeclaredType) tm;
-                    if(declaredType.getTypeArguments()!=null && declaredType.getTypeArguments().size()>0) {//it's some generic type
-                        if(mType == IMappingType.ONE_TO_MANY || mType == IMappingType.MANY_TO_MANY) {//we suppose it should be for relationship mapping only
-                            tm = declaredType.getTypeArguments().get(0);
-                            if(tm.getKind() == TypeKind.DECLARED){
-                                declaredType = (DeclaredType) tm;
+        } else {
+            tm = var.asType();
+        }
+        if( tm != null ){
+            if(null != tm.getKind()) {
```

Comment:
Would it make sense to pull this up into the upper if? To me these if -> switch changes don't improve readabiliy, but ok.

Reply:
Done.